src/util/shell.cpp

```diff
@@ -752,13 +752,6 @@ extern "C" LEAN_EXPORT int lean_main(int argc, char ** argv) {
             out.close();
         }
 
-        // target triple is only used by the LLVM backend. Save users
-        // a great deal of pain by erroring out if they misuse flags.
-        if (target_triple && !llvm_output) {
-            std::cerr << "ERROR: '--target' must be used with '--bc' to enable LLVM backend. Quitting code generation.\n";
-            return 1;
-        }
```

Comment:
Can you explain/recap the motivation for this change? We're not forwarding `--target` to `lean` right now, are we?

Reply:
We did notice that not doing this does trigger the warning in test runs, e.g. here: https://github.com/leanprover/lean4/actions/runs/5789866313/job/15691817010 so it does seem that it gets forwarded?

Reply:
Yes because it's reintroduced on that branch: https://github.com/leanprover/lean4/compare/master...bollu:lean4:kha-llvm-stage2-build#diff-0d112c28beff241f98979d3807483fed193e94d8e0e9db9df7d2f5bd878a6cb8R42-R43